cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/Context/SchemaManagerUtilities.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {
+  INode,
+  parseUnionType,
+  parseArrayType,
+  parseEnumType,
+  parseMapType,
+  IOrderedChildren,
+  parseComplexType,
+} from 'components/AbstractWidget/SchemaEditor/Context/SchemaParser';
+import {
+  logicalTypes,
+  defaultTimeStampTypeProperties,
+  defaultDecimalTypeProperties,
+  defaultTimeTypeProperties,
+  defaultDateTypeProperties,
+  defaultArrayType,
+  defaultEnumType,
+  defaultMapType,
+  defaultRecordType,
+  defaultUnionType,
+  InternalTypesEnum,
+} from 'components/AbstractWidget/SchemaEditor/SchemaConstants';
+import isEmpty from 'lodash/isEmpty';
+
+function getInternalType(tree: INode) {
+  const hasChildren = tree.children ? Object.keys(tree.children).length : 0;
```

Comment:
nit: The naming of this (currently numeric) variable sounds like it is a boolean, and it looks like it is being used as a boolean value. Maybe it can be modified slightly to have a boolean value.